sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_receiver_async.py

```diff
@@ -103,14 +133,48 @@ async def _outgoing_disposition(
         settled: Optional[bool],
         state: Optional[Union[Received, Accepted, Rejected, Released, Modified]],
         batchable: Optional[bool],
+        *,
+        message: Optional["_MessageDelivery"] = None,
+        on_disposition: Optional[Callable] = None,
     ):
         disposition_frame = DispositionFrame(
             role=self.role, first=first, last=last, settled=settled, state=state, batchable=batchable
         )
+
         if self.network_trace:
             _LOGGER.debug("-> %r", DispositionFrame(*disposition_frame), extra=self.network_trace_params)
+
+        # If trying to settle a message, keep track of the disposition
+        if message:
+            delivery = PendingDisposition(
+                message = message,
+                on_delivery_settled = on_disposition,
+            )
+            delivery.frame = disposition_frame
+            delivery.settled = settled
+            delivery.transfer_state = state
+            delivery.start = time.time()
+            delivery.sent = True
```

Comment:
Any reason we setting all these as attributes rather than via constructor?

Reply:
I was modeling after the sender model -- will fix :)